### PR TITLE
Topic/more date ops

### PR DESCRIPTION
**Pull request by HuwCampbell**
This kind of abuses the operators, but the functionality is as it should be.

### CODE REVIEW

File: src/Icicle/Source/Parser/Token.hs [hunk 1]
```diff
@@ -53,7 +54,11 @@ pKeyword kw
 
 pOperator :: Parser T.Operator
 pOperator
- = pTok get <?> "operator"
+ = pTok get
+   <|> (pKeyword T.Days *> (T.Operator ("days before") <$ pKeyword T.Before <|> T.Operator ("days after") <$ pKeyword T.After))
+   <|> (pKeyword T.Weeks *> (T.Operator ("weeks before") <$ pKeyword T.Before <|> T.Operator ("weeks after") <$ pKeyword T.After))
+   <|> (pKeyword T.Months *> (T.Operator ("months before") <$ pKeyword T.Before <|> T.Operator ("months after") <$ pKeyword T.After))
+   <?> "operator"
```
Comment (tranma) [inline src/Icicle/Source/Parser/Token.hs]: seems ok, it's not that bad...

**Comment by tranma** [comment]
:+1: I think using "operators" in this way is ok...

**Comment by amosr** [comment]
:+1: repl test so we have an example of using it, too.
what happens if you put two spaces in - or line break?
``3 months   before now``
etc

**Comment by HuwCampbell** [comment]
Yep, I have actually been writing up window tests using it, but I'll do one in icicle proper's too.
No problem for multiple spaces or newline, the lexer skips just gives back the tokens.

**Comment by amosr** [comment]
oh man
```
3 months       before              now
```
I thought code mode would respect whitespace. weird

**Comment by amosr** [comment]
ok. double doinks do but triple doinks dont

**Pull request merged**